src/h5web/vis-packs/core/containers/HeatmapVisContainer.tsx

```diff
@@ -21,21 +25,34 @@ function HeatmapVisContainer(props: VisContainerProps) {
   const { name, shape: dims } = entity;
   const [dimMapping, setDimMapping] = useDimMappingState(dims, 2);
 
+  const { valuesStore } = useContext(ProviderContext);
+
   return (
     <>
       <DimensionMapper
         rawDims={dims}
         mapperState={dimMapping}
         onChange={setDimMapping}
       />
-      <Suspense fallback={<ValueLoader />}>
-        <MappedHeatmapVis
-          dataset={entity}
-          dims={dims}
-          dimMapping={dimMapping}
-          title={name}
-        />
-      </Suspense>
+      <ErrorBoundary
+        resetKeys={[dimMapping]}
+        FallbackComponent={ErrorFallback}
+        onReset={() => {
+          valuesStore.evict({
+            path: entity.path,
+            selection: getSliceSelection(dimMapping),
```

Comment:
This is how I evict a slice from the `valuesStore`'s cache.

Reply:
What happens if the error caught by the boundary is not a `RequestCancelled` error ? Will it also remove the value from the store on reset ?

Reply:
I only show the `retry` button when the error is a `RequestCancelled` error.

Reply:
Is this the only way to reset the error boundary from the UI ?

Reply:
What do you have in mind?

Reply:
Nothing in particular. I was just wondering if an user could remove the value from the store through a side-effect, which is not what we want.

Reply:
Right, I think I understand your concern: you're asking when `onReset` is called basically?

It only gets called when the boundary is reset manually by calling `resetErrorBoundary` inside it. When the boundary resets because one of the `resetKeys` passed to it changes, `onReset` is not called.

Does that make sense?

Reply:
It does ! It also lifts my concerns, thanks :slightly_smiling_face: